lib/msf/core/db_manager/ref.rb

```diff
@@ -5,6 +5,7 @@ module Msf::DBManager::Ref
   def find_or_create_ref(opts)
     ret = {}
     ret[:ref] = get_ref(opts[:name])
+    ret[:ref] = Mdm::Ref.find_by_id(opts[:id]) if opts[:id]
```

Comment:
Why does this perform a lookup by name, followed by a lookup by ID? If the name lookup returned a result it becomes redundant if the ID based lookup returns a result. In addition the `Mdm::Ref.find_by_id` call takes place outside of a `::ActiveRecord::Base.connection_pool.with_connection` block.

Reply:
A few of the `DBManager` modules have a pattern that simply calls `find` if `opts` contains an id key. Would that be a better option here? If it doesn't then it would perform the `get_ref` check?

Example from `lib/msf/core/db_manager/host.rb`:

```
# If we have the ID, there is no point in creating a complex query.
if opts[:id] && !opts[:id].to_s.empty?
  return Array.wrap(Mdm::Host.find(opts[:id]))
end
```

Reply:
The lookup by name is legacy behavior. Before my changes that was the only way to look up the a `Ref` through `DBManager`. I do agree that the `id` lookup should be before the `name` lookup and that we should match the style of the other methods. I'll make those changes.

Reply:
I updated the behavior in the latest push to match others. Just so you're aware, I originally wrapped the results in an array, because that is consistent with other DbManager operations, but I then realized this was breaking things (was missing a vuln when running my standard test script) so I reverted it back. I believe this is OK since we do not expose this endpoint via the API, so we won't have problems conforming to the data format standard.